helpers/hdocker.py

```diff
@@ -964,6 +964,104 @@ def run_dockerized_pandoc(
         ret = None
     return ret
 
+# #############################################################################
+# Dockerized extract_notebook_images.
+# #############################################################################
+
+def run_dockerized_notebook_image_extractor(
+    notebook_path: str,
+    output_dir: str,
+    *,
+    force_rebuild: bool = False,
+    use_sudo: bool = False,
+) -> None:
+    """
+    Run NotebookImageExtractor in a Docker container.
+
+    """
+    _LOG.debug(hprint.func_signature_to_str())
+    # Build the container image, if needed.
+    container_image = "tmp.notebook_image_extractor"
+    dockerfile = r"""
+    FROM --platform=linux/amd64 python:3.10-slim
+
+    # Install required system libraries for Chromium and Playwright.
+    RUN apt-get update && apt-get install -y \
+        libglib2.0-0 \
+        libnss3 \
+        libnspr4 \
+        libdbus-1-3 \
+        libatk1.0-0 \
+        libatk-bridge2.0-0 \
+        libexpat1 \
+        libatspi2.0-0 \
+        libdbus-glib-1-2 \
+        libxcomposite1 \
+        libxdamage1 \
+        libxfixes3 \
+        libxrandr2 \
+        libgbm1 \
+        libxkbcommon0 \
+        libasound2 \
+        libcups2 \
+        libpango-1.0-0 \
+        libcairo2 \
+        && rm -rf /var/lib/apt/lists/*
+
+    # Set the environment variable for Playwright to install browsers in a known location.
+    ENV PLAYWRIGHT_BROWSERS_PATH=/ms-playwright
+
+    # Create the directory for Playwright browsers and ensure it's writable.
+    RUN mkdir -p /ms-playwright && chmod -R 777 /ms-playwright
+
+    RUN pip install nbconvert nbformat playwright pyyaml
+
+    RUN python -m playwright install
+
+    WORKDIR /app
+    """
+    container_image = build_container_image(
+        container_image, dockerfile, force_rebuild, use_sudo
+    )
+    # Convert file paths to Docker paths.
+    is_caller_host = not hserver.is_inside_docker()
+    use_sibling_container_for_callee = True
+    caller_mount_path, callee_mount_path, mount = get_docker_mount_info(
+        is_caller_host, use_sibling_container_for_callee
+    )
+    notebook_path = convert_caller_to_callee_docker_path(
+        notebook_path,
+        caller_mount_path,
+        callee_mount_path,
+        check_if_exists=True,
+        is_input=True,
+        is_caller_host=is_caller_host,
+        use_sibling_container_for_callee=use_sibling_container_for_callee,
+    )
+    output_dir = convert_caller_to_callee_docker_path(
+        output_dir,
+        caller_mount_path,
+        callee_mount_path,
+        check_if_exists=False,
+        is_input=False,
+        is_caller_host=is_caller_host,
+        use_sibling_container_for_callee=use_sibling_container_for_callee,
+    )
+    bash_cmd = (
```

Comment:
can we build it with f-strings and in a more readable way

python_code = ...
bash_cmd = "python -c '{python_code}'"